---
title: "Main analysis"
author: "ARUL"
date: "10/12/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(tidyverse)
library(dplyr)
library(ggplot2)
library(dplyr)
library(edgeR)
library(limma)
library(UpSetR)
library(flashClust)
library(GEOquery)
library(dendextend)
library(RColorBrewer)
library(knitr)
library(kableExtra)
library(FactoMineR)
```

1. Importing the datasets to R 
```{r}
library(readxl)
data1<-read.csv(file = "M2PHDS_19-20_OMICS_CLIN_DATA_MAARS_AD_full_20190131_12-34-49 (1).csv", header = TRUE, sep = "\t")
data2 <- read_excel("M2PHDS_19-20_OMICS_CLIN_DATA_MAARS_all_Fri_Apr_04_14h_CEST_2014.xlsx")
data3<- read.delim("M2PHDS_19-20_OMICS_TRANSC_MAARS_normTranscriptome_618samples_16042014.txt")
```

2. Taking all the participants that have gene expression from data 1 and data2 through MAARS identifier
```{r, filtering the samples that have only gene expression}
data2 <- data2 %>% filter(sample_id %in% colnames(data3))
data1 <- data1 %>% filter(data1$patient.Identification.MAARS.identifier..MAARS_identifier. %in% annotations$MAARS_identifier) 

  
data2_AD<- data2 %>% filter(clinical_group == "AD")
data2_CTRL<- data2 %>% filter(clinical_group == "CTRL")
data2_PSO <- data2 %>% filter(clinical_group == "PSO")

# leaving the PSO data from the analysis

data2_AD_CTRL <- data2 %>% filter(clinical_group == "AD" | clinical_group == "CTRL")
data2_AD_Lesional <- data2 %>% filter(clinical_group == "AD" | lesional == "LES")


```


```{r, linking and merging the the three main dataframes with required columns by join function}
# taking the participant
sample_id <- colnames(data3)

annotations <- data.frame(sample_id = colnames(data3), MAARS_identifier = gsub('.{3}$', '', sample_id))

annotations <- dplyr::left_join(annotations, data2, by = "sample_id") %>% select(sample_id, MAARS_identifier.x, clinical_group, lesional, CUSTOM_Age, Gender, Institution) %>% rename(MAARS_identifier = MAARS_identifier.x)

# Categorising thhe SCORAD into three levels - mild, moderate and severe
SCORAD_cat<- cut(data1$patient.SCORAD.index.SCORAD.SCORAD.Score..SCORAD_Score., breaks = c(0, 25, 50, 100), labels = c("Mild", "Moderate", "Severe"), right = FALSE)
data1 <- cbind(data1, SCORAD_cat)


Infos_dataset <- dplyr::left_join(annotations, data1, by = c("sample_id" = "involved.skin.biopsy.involved.skin.biopsy.MAARS.Sample.identifier..MAARS_Sample_identifier.")) %>% select(sample_id, MAARS_identifier, clinical_group, lesional, CUSTOM_Age, Gender, Institution, SCORAD_cat)%>% rename(SCORAD_status = SCORAD_cat)


# Making the control group has  No_Atopic dermatitis in SCORAD status

levels<- levels(Infos_dataset$SCORAD_status)
levels[length(levels) + 1] <- "No_Atopic_dermatitis"

Infos_dataset$SCORAD_status <- factor(Infos_dataset$SCORAD_status, levels = levels)
Infos_dataset$SCORAD_status[is.na(Infos_dataset$SCORAD_status)] <- "No_Atopic_dermatitis"

class(Infos_dataset$SCORAD_status)
  
#removal of PSO from the analysis as we are going to focus on the AD lesional and non lesional in comparison with the Ctrl group.

Infos_dataset_AD_CTRL <- Infos_dataset %>% filter(clinical_group == "AD" | clinical_group == "CTRL")

#Identifying only AD and CTRL to the data3(gene expression dataset)

Geneset_AD_CTRL <- data3[, colnames(data3) %in% Infos_dataset_AD_CTRL$sample_id]

```


```{r}

# Creating the additional variable for participant group belonging to AD lesional, non-lesional and for control seperately
Infos_dataset_AD_CTRL$Participant_group[Infos_dataset_AD_CTRL$clinical_group == "AD" & Infos_dataset_AD_CTRL$lesional == "LES"] <- "AD_lesional"
Infos_dataset_AD_CTRL$Participant_group[Infos_dataset_AD_CTRL$clinical_group == "AD" & Infos_dataset_AD_CTRL$lesional == "NON_LES"] <- "AD_nonlesional"
Infos_dataset_AD_CTRL$Participant_group[Infos_dataset_AD_CTRL$clinical_group == "CTRL"] <- "CTRL"


Infos_dataset_AD_CTRL$clinical_group <- as.factor(Infos_dataset_AD_CTRL$clinical_group)
Infos_dataset_AD_CTRL$lesional <- as.factor(Infos_dataset_AD_CTRL$lesional)
Infos_dataset_AD_CTRL$Participant_group<- as.factor(Infos_dataset_AD_CTRL$Participant_group)
Infos_dataset_AD_CTRL$Institution <- as.factor(Infos_dataset_AD_CTRL$Institution)

class(Infos_dataset_AD_CTRL$Institution)

data2_AD_CTRL$Participant_group[data2_AD_CTRL$clinical_group == "AD" & data2_AD_CTRL$lesional == "LES"] <- "AD_lesional"
data2_AD_CTRL$Participant_group[data2_AD_CTRL$clinical_group == "AD" & data2_AD_CTRL$lesional == "NON_LES"] <- "AD_nonlesional"
data2_AD_CTRL$Participant_group[data2_AD_CTRL$clinical_group == "CTRL"] <- "CTRL"

# releveling the datset

Infos_dataset_AD_CTRL$clinical_group <- relevel(Infos_dataset_AD_CTRL$clinical_group, ref = "CTRL")
Infos_dataset_AD_CTRL$lesional <- relevel(Infos_dataset_AD_CTRL$lesional, ref = "NON_LES")
Infos_dataset_AD_CTRL$Participant_group <- relevel(Infos_dataset_AD_CTRL$Participant_group, ref = "CTRL")
Infos_dataset_AD_CTRL$SCORAD_status <- relevel(Infos_dataset_AD_CTRL$SCORAD_status, ref = "No_Atopic_dermatitis")

#Constructing the Digital gene expression sheet for all the samples
library(Biobase)

assayData = (as.matrix(Geneset_AD_CTRL))

Expression_set_AD_CTRL<- ExpressionSet(assayData = (as.matrix(Geneset_AD_CTRL)), phenoData = (annotatedDataFrameFrom(data.frame(data2_AD_CTRL, row.names = 1, header= TRUE))))




```

## Descriptive analysis  
 
 
```{r, gender}
library(tidyverse)
library(knitr)



kable(data2_AD_CTRL %>% group_by(data2_AD_CTRL$clinical_group) %>% filter(MAARS_identifier != "MAARS_1_017")%>%
  summarise(n_distinct = n_distinct(MAARS_identifier), mean_age = mean(CUSTOM_Age), sd_age = sd(CUSTOM_Age),
            percent_female = sum(Gender == "Female")/n()*100, family_history_AD = sum(CUSTOM_Fam._hist._Atopic_dermatitis == TRUE, na.rm = TRUE)/n()*100, family_history_PSO = sum(CUSTOM_Family_History_of_Psoriasis == TRUE, na.rm = TRUE)/n()*100))


kable(data2_AD_CTRL %>% group_by(data2_AD_CTRL$clinical_group) %>% filter(MAARS_identifier != "MAARS_1_017")%>%
  summarise(n_distinct = n_distinct(MAARS_identifier), allergy_pseudo_drug = sum(Known_Allergies_v2..Pseudo_Drug_Allergy == TRUE)/n()*100, allergy_dustmite = sum(Known_Allergies_v2..House_dust_mite == TRUE)/n()*100, allergy_food = sum(Known_Allergies_v2..Food == TRUE)/n()*100, allergy_pollen = sum(Known_Allergies_v2..Pollen == TRUE)/n()*100, allergy_contact = sum(Known_Allergies_v2..Contact_Allergy == TRUE)/n()*100, allergy_drug = sum(Known_Allergies_v2..Drug_Allergy == TRUE)/n()*100, allergy_animal = sum(Known_Allergies_v2..Animal == TRUE)/n()*100))


kable(data2_AD_CTRL %>% group_by(clinical_group)%>% filter(MAARS_identifier != "MAARS_1_017")%>%
        summarise(n_distinct = n_distinct(MAARS_identifier), Prop_UB = sum(anatomical_location_label == 'UB')/n()*100, Prop_PT = sum(anatomical_location_label == 'PT')/n()*100, Prop_LB = sum(anatomical_location_label == 'LB')/n()*100, Prop_T = sum(anatomical_location_label == "T")/n()*100, Prop_B = sum(anatomical_location_label == "B")/n()*100))


kable(data2_AD_CTRL %>% group_by(clinical_group)%>% filter(MAARS_identifier != "MAARS_1_017")%>%
        summarise(n_distinct = n_distinct(MAARS_identifier), percentage_HHU = sum(Institution == "HHU")/n()*100, percentage_KINGS = sum(Institution == "KINGS")/n()*100, percentage_UH = sum(Institution == "UH")/n()*100))


```


```{r, age and clinical group}
#Two sampe t test
t.test(CUSTOM_Age ~ clinical_group, data = Infos_dataset_AD_CTRL, var.equal = TRUE)
# Chi-square test

chisq.test(data2_AD_CTRL$clinical_group, data2_AD_CTRL$lesional, correct = FALSE)

chisq.test(data2_AD_CTRL$clinical_group, data2_AD_CTRL$Gender, correct = FALSE)

chisq.test(data2_AD_CTRL$clinical_group, data2_AD_CTRL$Institution, correct = FALSE)

chisq.test(data2_AD_CTRL$clinical_group, data2_AD_CTRL$anatomical_location_label, correct = FALSE)

chisq.test(data2_AD_CTRL$clinical_group, data2_AD_CTRL$Institution, correct = FALSE)

chisq.test(data2_AD_CTRL$clinical_group, data2_AD_CTRL$CUSTOM_Fam._hist._Atopic_dermatitis, correct = FALSE)

chisq.test(data2_AD_CTRL$clinical_group, data2_AD_CTRL$CUSTOM_Fam._hist._Atopic_dermatitis)

chisq.test(data2_AD_CTRL$clinical_group, data2_AD_CTRL$Known_Allergies_v2..Pseudo_Drug_Allergy)

chisq.test(data2_AD_CTRL$clinical_group, data2_AD_CTRL$Known_Allergies_v2..House_dust_mite)

chisq.test(data2_AD_CTRL$clinical_group, data2_AD_CTRL$Known_Allergies_v2..Food)

chisq.test(data2_AD_CTRL$clinical_group, data2_AD_CTRL$Known_Allergies_v2..Pollen)

chisq.test(data2_AD_CTRL$clinical_group, data2_AD_CTRL$Known_Allergies_v2..Contact_Allergy)

chisq.test(data2_AD_CTRL$clinical_group, data2_AD_CTRL$Known_Allergies_v2..Drug_Allergy)

chisq.test(data2_AD_CTRL$clinical_group, data2_AD_CTRL$Known_Allergies_v2..Animal)

```



```{r, Age}
kable(Infos_dataset_AD_CTRL %>% group_by(Participant_group)%>% summarise( mean_age = mean(CUSTOM_Age), sd_age = sd(CUSTOM_Age)))

#taking out control samples from the group


kable(Infos_dataset_AD_CTRL %>% group_by(Gender)%>% summarise(Control = mean(Participant_group == "CTRL")/n()*100, AD_lesional = mean(Participant_group == "AD_lesional")/n()*100, AD_non_lesional = mean(Participant_group == "AD_nonlesional")/n()*100))
```


```{r, Institution}
kable(Infos_dataset_AD_CTRL %>% group_by(Participant_group)%>% summarise(percentage_HHU = sum(Institution == "HHU")/n()*100, percentage_KINGS = sum(Institution == "KINGS")/n()*100, percentage_UH = sum(Institution == "UH")/n()*100))
```

```{r, SCORAD SCORE}
kable(Infos_dataset_AD_CTRL  %>% group_by(Participant_group)%>% summarise(Prop_mild = sum(SCORAD_status == "Mild")/n()*100, Prop_moder = sum(SCORAD_status == "Moderate")/n()*100, Prop_Severe = sum(SCORAD_status == "Severe")/n()*100))

```

```{r, anatomical location}
kable(data2_AD_CTRL %>% group_by(Participant_group)%>% summarise(Prop_UB = sum(anatomical_location_label == 'UB')/n()*100, Prop_PT = sum(anatomical_location_label == 'PT')/n()*100, Prop_LB = sum(anatomical_location_label == 'LB')/n()*100, Prop_T = sum(anatomical_location_label == "T")/n()*100, Prop_B = sum(anatomical_location_label == "B")/n()*100))

```

```{r}
kable(data2_AD_CTRL  %>% group_by(Participant_group)%>% summarise(Prop_mild = sum(SCORAD_status == "Mild")/n()*100, Prop_moder = sum(SCORAD_status == "Moderate")/n()*100, Prop_Severe = sum(SCORAD_status == "Severe")/n()*100))
```


```{r, Ethnicity}
kable(data1 %>% summarise(percentage_white = sum(patient.Diagnostic...Phenotypic.Data.Ethnicity.Family.History.Ethnicity..Ethnicity. == "White")/n()*100, percentage_Asian_others = sum(patient.Diagnostic...Phenotypic.Data.Ethnicity.Family.History.Ethnicity..Ethnicity. == "Asian - Other")/n()*100,percentage_Black_african = sum(patient.Diagnostic...Phenotypic.Data.Ethnicity.Family.History.Ethnicity..Ethnicity. == "Black - African")/n()*100, percentage_Black_carribean = sum(patient.Diagnostic...Phenotypic.Data.Ethnicity.Family.History.Ethnicity..Ethnicity. == "Black - Caribbean")/n()*100, percentage_Black_others = sum(patient.Diagnostic...Phenotypic.Data.Ethnicity.Family.History.Ethnicity..Ethnicity. == "Black - Other")/n()*100))

table(data1$patient.Diagnostic...Phenotypic.Data.Ethnicity.Family.History.Ethnicity..Ethnicity.)
```

```{r, Allergy}
kable(data1 %>%  
  summarise(Family_history_True = sum(patient.Diagnostic...Phenotypic.Data.Ethnicity.Family.History.Family.History.of.atopic.diseases..Family_History_of_atopic_diseases. == TRUE)/n()*100, Family_history_False == sum(patient.Diagnostic...Phenotypic.Data.Ethnicity.Family.History.Family.History.of.atopic.diseases..Family_History_of_atopic_diseases. == FALSE)/n()*100))


# family history of AD
kable(data2_AD_CTRL %>%  group_by(clinical_group)%>%
  summarise(Family_historyAD_True = sum(CUSTOM_Fam._hist._Atopic_dermatitis == TRUE)/n()*100, Family_historyAD_false = sum(CUSTOM_Fam._hist._Atopic_dermatitis == FALSE)/n()*100))




# Allergies
kable(data2_AD_CTRL%>% group_by(clinical_group)%>% summarise(Pseudodrug_Allergiestrue = sum(Known_Allergies_v2..Pseudo_Drug_Allergy == TRUE)/n()*100, allergy_Animal = sum(Known_Allergies_v2..Animal  == TRUE)/n()*100, allergy_contact = sum(Known_Allergies_v2..Contact_Allergy == TRUE)/n()*100, allergy_drug = sum(Known_Allergies_v2..Drug_Allergy == TRUE)/n()*100, allergy_food = sum(Known_Allergies_v2..Food == TRUE)/n()*100, allergy_pollen = sum(Known_Allergies_v2..Pollen == TRUE)/n()*100, allergy_house = sum(Known_Allergies_v2..House_dust_mite == TRUE)/n()*100))


```



# Diagnostic plots

MA plot

```{r, echo=TRUE}
BiocManager::install("DESeq2")
library(DESeq2)


dds<- DESeq(Geneset_AD_CTRL)


MA_plot<- limma::plotMA(object = Geneset_AD_CTRL, ylim = c(-3, 3), hl.col = c("blue", "red"), main = " MA plot of Expressed genes among AD and control ")
abline(h = 0)


plotMA(Geneset_AD_CTRL, array=250, status=Infos_dataset_AD_CTRL$clinical_group, hl.col = c("blue", "red"), main = " MA plot of Expressed genes among AD and control ")
abline(h =0)

plotWithHighlights(Geneset_AD_CTRL, Infos_dataset_AD_CTRL$clinical_group, status = NULL, values = NULL,
                   hl.pch = 16, hl.col = NULL, hl.cex = 1, legend = "topright",
                   bg.pch = 16, bg.col = "black", bg.cex = 0.3,
                   pch = NULL, col = NULL, cex = NULL)


```



```{r, eval}
# Constructing the Hierarchial clustering dendrogram

  deucl_norm <- dist(t(Geneset_AD_CTRL))
plot(deucl_norm)


plot(flashClust::hclust(deucl_norm, method = "ward"), labels = Infos_dataset_AD_CTRL$Participant_group, col = Infos_dataset_AD_CTRL$clinical_group)


# Heat map contruction by base r package
t_geneset_AD_CTRL<- t(Geneset_AD_CTRL)
heatmap(t_geneset_AD_CTRL, scale = "none")

#  High values are in red and low values are in yellow.
```

## DGE and modeling matrix


```{r, eval}
library(limma)

# modelling the matrix through limma package as the RNA micro array data is a continous variable

design <- model.matrix(~ Infos_dataset_AD_CTRL$clinical_group)
dim(design)
y<- voom(Geneset_AD_CTRL, design, plot = T)


fit<- lmFit(Geneset_AD_CTRL, design)

Bayesfit<- eBayes(fit, robust = TRUE)

voomlimma_signif <- decideTests(Bayesfit, adjust.method = "BH", p.value = 0.05, lfc = 1) # as we previously described that Benjamin hochberg method to chose accounting for a compromise and not to be conservative
summary(voomlimma_signif)
plotSA((Bayesfit))
voomlimma_signif[]
topTable(Bayesfit, adjust.method = "BH", number = Inf)
summary(Bayesfit)


```


# Performing the MDS 
```{r}
# Classical MDS
# N rows (objects) x p columns (variables)
# each row identified by a unique row name

d <- dist(Geneset_AD_CTRL) # euclidean distances between the rows
fit_CMD <- cmdscale(d,eig=TRUE, k=2) # k is the number of dim
      fit_CMD # view results

# plot solution
x <- fit$points[,1]
y <- fit$points[,2]
plot(x, y, xlab="Coordinate 1", ylab="Coordinate 2",
  main="Metric MDS", type="n")
text(x, y, labels = row.names(mydata), cex=.7)


```



```{r}
# keeping the confidence interval as 95% for the fold changes
topTable(Bayesfit, coef = 'lesional')

res <- topTable(Bayesfit, coef="lesional", confint=0.95)

keep <- res$CI.L > -1 & res$CI.R < 1
res[keep,]
topTable(Bayesfit, adjust="BH", p.value = 0.05, lfc = 0.5849, coef = "lesional")
```


What is voom doing?

Counts are transformed to log2 counts per million reads (CPM), where “per million reads” is defined based on the normalization factors we calculated earlier
A linear model is fitted to the log2 CPM for each gene, and the residuals are calculated
A smoothed curve is fitted to the sqrt(residual standard deviation) by average expression (see red line in plot above)
The smoothed curve is used to obtain weights for each gene and sample that are passed into limma along with the log2 CPMs.

```{r, eval}
#Multiple testing to see the significance levels for different method 
pval_ord <- cbind.data.frame("raw_pval" = sort(Bayesfit$p.value[,2]), "order" =1:nrow(Geneset_AD_CTRL))
pval_ord <- cbind.data.frame("raw_pval" = sort(fit$p.value[,2]), "order" =1:nrow(Geneset_AD_CTRL))

myalpha <- 0.05

table(pval_ord$raw_pval < myalpha)

table(p.adjust(pval_ord$raw_pval, method = "BH") < myalpha)

table(p.adjust(pval_ord$raw_pval, method = "holm") < myalpha)

table(p.adjust(pval_ord$raw_pval, method = "bonferroni") < myalpha)


m <- nrow(pval_ord)
pval_ord$bh <- myalpha/m * 1:m
pval_ord$holm <- myalpha/(m:1)
pval_ord$nocor <- rep(myalpha, m)
pval_ord$bonfer <- rep(myalpha/m, m)

# Plotting the different method p-values for examining the siginficance point
install.packages("ggplot2")

library(ggplot2)
ggplot(pval_ord, aes(x = order)) + geom_point(aes(y = raw_pval), shape = 1) + 
    scale_y_log10(minor_breaks = rep(1:9, 13) * 10^-rep(0:12, each = 9)) + geom_line(aes(y = bh, 
    col = "Benjamini-Hochberg")) + geom_line(aes(y = holm, col = "Holm")) + 
    geom_line(aes(y = bonfer, col = "Bonferroni")) + geom_line(aes(y = nocor, 
    col = "No correction")) + theme_classic() + theme(panel.grid.minor.x = element_blank(), 
    panel.grid.major.x = element_blank()) + scale_colour_viridis_d("Correction limit") + 
    xlab("P-value rank (ascending order)") + ylab("Moderated empirical Bayes t-test p-value (log10 scale)")


```

```{r}
# plotting the dimension plotting through limma package

plotMDS(list1, top = 500, dim.plot = c(1,2), ndim = max(dim.plot), gene.selection = "pairwise", xlab = "Dimension 1", ylab = "Dimension 2", plot = TRUE)

# but it is not working, have to figure out
```

```{r}
# Trying out the sPLS method to check for the dimensions and variance explained by each dimensions

BiocManager::install("mixOmics")

library(mixOmics)
x<- t(data3)

y<- subset(annotations1, Institution == "UH", SCORAD_status)$SCORAD_status

y <- as.factor(y)
p <- ncol(x)
n<- nrow(x)
spls1<-splsda(x, y, ncomp = 10, keepX = rep(5, 10))
# error throws unequal no of rows in x and y, have to solve it
```




```{r}
# PCA procedure

library(FactoMineR)

AD_CTRL_PCA <- PCA(t_geneset_AD_CTRL)

#Clinical group

AD_CTRL_Genes_Clinical_group<- data.frame(t_geneset_AD_CTRL, location = Infos_dataset_AD_CTRL$clinical_group)


PCA_AD_CTRL_Clinical_group <- PCA(AD_CTRL_Genes_Clinical_group, quali.sup = which(names(AD_CTRL_Genes_Clinical_group) == 
    "location"))

plot(PCA_AD_CTRL_Clinical_group, choix = "ind", label = "none", habillage = which(names(AD_CTRL_Genes_Clinical_group) == 
    "location"), addEllipses=TRUE)

#Lesional vs Non lesional

 AD_CTRL_Genes_Lesional<- data.frame(t_geneset_AD_CTRL, location = Infos_dataset_AD_CTRL$lesional)

rawPCAcolored1 <- PCA(AD_CTRL_Genes_Lesional, quali.sup = which(names(AD_CTRL_Genes_Lesional) == 
    "location"))

plot(rawPCAcolored1, choix = "ind", label = "none", habillage = which(names(AD_CTRL_Genes_Lesional) == 
    "location"))


  # Scorad score

rawCommonGenes2 <- data.frame(t_geneset_AD_CTRL, location = Infos_dataset_AD_CTRL$SCORAD_status)

rawPCAcolored2 <- PCA(rawCommonGenes2, quali.sup = which(names(rawCommonGenes2) == 
    "location"))

plot(rawPCAcolored2, choix = "ind", label = "none", habillage = which(names(rawCommonGenes2) == 
    "location"))
```


```{r}
#Scree plot for examining the proportion of variance in a dataset that is explained by each of the components in PCA

eigenvalues<- PCA_AD_CTRL_Clinical_group$eig

barplot(eigenvalues[, 2], names.arg=1:nrow(eigenvalues), 
       main = "Scree Plot",
       xlab = "Principal Components",
       ylab = "Percentage of variances",
       col ="steelblue")
# Add connected line segments to the plot
lines(x = 1:nrow(eigenvalues), eigenvalues[, 2], 
      type="b", pch=19, col = "red")


```

```{r}
# Visualizing the variable plots in PCA

fviz_pca_var(PCA_AD_CTRL_participan_group, col.var = Infos_dataset_AD_CTRL$Participant_group, 
             palette = c("#0073C2FF", "#EFC000FF", "#868686FF"),
             legend.title = "Cluster")
```



#Adding ellipses to PCA plot
```{r}
plotellipses(rawPCAcolored1, keepvar = "all" )
plotellipses(rawPCAcolored1, keepvar = "all", axes = c(1, 2), means=TRUE, level = 0.95, 
    magnify = 2, cex = 1, pch = 20, pch.means=15, type = c("g","p"), 
    keepnames = TRUE, namescat = NULL, xlim=NULL, ylim=NULL, lwd=1, 
    label="all", autoLab=c("auto","yes","no"))


library("devtools")
install_github("kassambara/factoextra")
library(factoextra)

fviz_pca_ind(rawPCAcolored1)

fviz_pca_ind(rawPCAcolored1, axes = c(1, 2), geom = c("point", "text"),
       label = "all", invisible = "none", labelsize = 4,
       pointsize = 2, habillage = "none",
       addEllipses = TRUE, ellipse.level = 0.95, 
       col.ind = "black", col.ind.sup = "blue", alpha.ind = 1,
       select.ind = list(name = NULL, cos2 = NULL, contrib = NULL),
       jitter = list(what = "label", width = NULL, height = NULL))


# Control automatically the color of individuals
# using the cos2 or the contributions
# cos2 = the quality of the individuals on the factor map
fviz_pca_ind(rawPCAcolored1, col.ind="cos2")


# PCA by color radient in cos2 value  Change the theme and use only points
fviz_pca_ind(rawPCAcolored1, col.ind="cos2", geom = "point") +
      scale_color_gradient2(low="white", mid="blue",
      high="red", midpoint=0.6)+ theme_minimal()

# Variable plot for PCA
fviz_pca_var(PCA_AD_CTRL_Clinical_group, col.var = "contrib",
             gradient.cols = c("#00AFBB", "#E7B800", "#FC4E07"))
fviz_pca_var(PCA_AD_CTRL_Clinical_group, col.var = "contrib", lim.cos2.var = 0.85)

fviz_pca_var(PCA_AD_CTRL_participan_group, col.var = "cos2", lim.cos2.var = 0.85)

fviz_pca_var(PCA_AD_CTRL_Clinical_group, select.var = list(cos2 = 0.85))

which.max(PCA_AD_CTRL_Clinical_group$var$cos2)
which.min(PCA_AD_CTRL_Clinical_group$var$cos2)

# correlation plot 
install.packages("corrplot")
library("corrplot")

var <- get_pca_var(rawPCAcolored1)
corrplot(var$cos2, is.corr=FALSE)


#eigen values
library(factoextra)
eig.val <- get_eigenvalue(rawPCAcolored1)
eig.val

fviz_eig(rawPCAcolored1, addlabels = TRUE, ylim = c(0, 50))

#Scree plot

fviz_eig(rawPCAcolored1, addlabels = TRUE, ylim = c(0, 10), ncp = 30, barcolor = "blue", barfill = "blue", ggtheme = theme_classic())
abline(h = 2)

plot(rawPCAcolored1$eig[, "eigenvalue"], type = "b", ylab = "Eigenvalue")
abline(h = ncol(AD_CTRL_Genes_Clinical_group)/nrow(rawPCAcolored1$eig), col = "red", lty = "dotted", lwd = 2)

fviz_eig(rawPCAcolored1, addlabels = TRUE, ylim = c(0, 10), ncp = 20, barcolor = "blue", barfill = "blue", ggtheme = theme_classic())



# To visualize the contribution

fviz_contrib(rawPCAcolored1, choice = "ind", axes = 1:2)

# Color by the contributions
fviz_pca_ind(rawPCAcolored1, col.ind="contrib") +
      scale_color_gradient2(low="white", mid="blue",
      high="red", midpoint=4)

# Color individuals by groups
fviz_pca_ind(rawPCAcolored1, label="none", habillage=Infos_dataset_AD_CTRL$clinical_group)
fviz_pca_ind(rawPCAcolored1, label="none", habillage=Infos_dataset_AD_CTRL$SCORAD_status)
fviz_pca_ind(PCA_AD_CTRL_participan_group, label = "none", habillage = Infos_dataset_AD_CTRL$Participant_group)

# Adding ellipses to the variable SCORAD status group
p <- fviz_pca_ind(rawPCAcolored1, label="none", habillage=Infos_dataset_AD_CTRL$SCORAD_status,
             addEllipses=TRUE, ellipse.level=0.95)
print(p)

p + scale_color_brewer(palette="Paired") +
     theme_minimal()

p + scale_color_brewer(palette="Set1") +
     theme_minimal()

# Adding ellipses to the variable Clinical group
q <- fviz_pca_ind(rawPCAcolored1, label="none", habillage=Infos_dataset_AD_CTRL$clinical_group,
             addEllipses=TRUE, ellipse.level=0.95)
print(q)

q + scale_color_brewer(palette="Paired") +
     theme_minimal()

q + scale_color_brewer(palette="Set1") +
     theme_minimal()
```





```{r, Creating a funciton as add_legend}
add_legend <- function(...) {
  opar <- par(fig=c(0, 1, 0, 1), oma=c(0, 0, 0, 0), 
    mar=c(0, 0, 0, 0), new=TRUE)
  on.exit(par(opar))
  plot(0, 0, type='n', bty='n', xaxt='n', yaxt='n')
  legend(...)
}
```




### Barplot of the library sizes (Checking the differences among clinical group(AD, PSO, CTRL))

Before moving on to doing the actually differential expression analysis it important do assess the quality of our data.

First, we can plot how many reads we have for each sample. Whilst normalisation can account for imbalance in coverage across the samples, extreme differences may be indicative of underlying problems in the samples.


```{r}
librarySizes <- colSums(Geneset_AD_CTRL)
# Clinical group wise examination of library sizes
barplot(librarySizes, 
    col = Infos_dataset_AD_CTRL$clinical_group, names = names(librarySizes), las=2, names.arg = "", ylim = c(145000, 150000), main="Barplot of library sizes" )
legend("topright", c("Atopic Dermatits", "Healthy Volunteers"), col = c("red", 
    "black"), pch = 15, horiz = FALSE, bg = NA)
abline(h = mean(librarySizes), lty = 2)

```

```{r}
# institution wise examination of library sizes
levels(Infos_dataset_AD_CTRL$Institution) <- c("blue2", "red", "green")
barplot(librarySizes, col = as.character(Infos_dataset_AD_CTRL$Institution), las=2, names.arg = "", ylim = c(145000, 150000), main="Barplot of library sizes" )
legend("topright", c("UH", "KINGS", "HHU"), col = c("blue2", "red", "green"), pch = 15, horiz = FALSE, bg = NA)
abline(h = mean(librarySizes), lty = 2)



```


# We’ll check the distribution of read counts using a boxplot and well add some colour to see if there is any difference between sample groups.



  When counts are high we approach a continuos signal, similar to a microarray experiment, and estimates will also be very precise. For abundantly expressed genes and with deep sequencing coverage, the biological variability will therefore dominate over sampling errors. In this case it should be perfectly fine to use standard microarray tools (which typically assume the data is normally distributed) on the normalized data.


```{r}
# make a colour vector
statusCol <- as.numeric(factor(Infos_dataset_AD_CTRL$clinical_group)) + 1
statuscol1<- as.numeric(factor(Infos_dataset_AD_CTRL$Institution)) + 1

# Check distributions of samples using boxplots
boxplot(Geneset_AD_CTRL, 
        xlab="", 
        ylab="RNA seq continous data",
        las=2,
        col=statusCol)
# Let's add a blue horizontal line that corresponds to the median logCPM
abline(h=median(as.matrix(Geneset_AD_CTRL)), col="blue")
add_legend("topright", legend = c("Atopic Dermatitis", "Control"), col = c("green", "red"), pch = 15, horiz = TRUE, bty = "n", cex = 0.8)

# making boxplot for instituition wise
statuscol1<- as.numeric(factor(Infos_dataset_AD_CTRL$Institution)) + 1

boxplot(Geneset_AD_CTRL, 
        xlab="", 
        ylab="RNA seq continous data",
        las=2,
        col=statuscol1)
# Let's add a blue horizontal line that corresponds to the median logCPM
abline(h=median(as.matrix(Geneset_AD_CTRL)), col="blue")
add_legend("topright", legend = c("UH", "KINGS", "HHU"), col = c("blue2", "red", "green"), pch = 15, horiz = TRUE, bty = "n", cex = 0.8)



```


#heatmap different color tryout
```{r}
install.packages("gplots")
library("gplots")
heatmap.2(t_geneset_AD_CTRL, scale = "none", trace = "none", density.info = "none")
```

# trying PCA in participant group with AD lesional, nonlesional and control group
```{r}
AD_CTRL_Genes_participant_group<- data.frame(t_geneset_AD_CTRL, location = Infos_dataset_AD_CTRL$Participant_group)

PCA_AD_CTRL_participan_group <- PCA(AD_CTRL_Genes_participant_group, quali.sup = which(names(AD_CTRL_Genes_participant_group) == 
    "location"))

 p1 <- fviz_pca_ind(PCA_AD_CTRL_participan_group, label="none", habillage=Infos_dataset_AD_CTRL$Participant_group,
+                   addEllipses=TRUE, ellipse.level=0.95)
 

```


